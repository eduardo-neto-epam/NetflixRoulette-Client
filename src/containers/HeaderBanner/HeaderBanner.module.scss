.bg-image {
    height: 500px;
    background:  linear-gradient(0deg, rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.6)), url(../../assets/myke-simon-atsUqIm3wxo-unsplash-min.jpg);
    background-size: cover;
    background-repeat: no-repeat;
    filter: blur(8px);
    transform: scale(1.03);
    flex-grow: 0;
}

.image-frame {
    overflow: hidden;
}

.container {
    position: absolute;
    top: 0;
    left: 50%;
    width: 100vw;
    padding: 40px;
    height: 600px;
    max-width: 1600px;
    transform: translateX(-50%);
}
