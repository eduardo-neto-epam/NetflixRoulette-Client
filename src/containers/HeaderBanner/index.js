import React from 'react'
import styles from './HeaderBanner.module.scss'
import classnames from 'classnames/bind'
import Header from '../Header'
import Heading from '../../components/Heading'
import SearchForm from '../../components/SearchForm'

let cx = classnames.bind(styles)

const HeaderBanner = () => {
    return (
        <>
        <div className={cx('image-frame')}>
            <div className={cx('bg-image')}></div>
        </div>
        <div className={cx('container')}>
            <Header />
            <Heading />
            <SearchForm />
        </div>
        </>
    )
}

export default HeaderBanner