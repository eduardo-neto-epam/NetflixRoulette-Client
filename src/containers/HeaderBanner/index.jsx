import React from 'react';
import classnames from 'classnames/bind';
import styles from './HeaderBanner.module.scss';
import Header from '../Header/index.jsx';
import Heading from '../../components/Heading/index.jsx';
import SearchForm from '../../components/SearchForm/index.jsx';

const cx = classnames.bind(styles);

const HeaderBanner = () => (
  <>
    <div className={cx('image-frame')}>
      <div className={cx('bg-image')} />
    </div>
    <div className={cx('container')}>
      <Header />
      <Heading />
      <SearchForm />
    </div>
  </>
);

export default HeaderBanner;
