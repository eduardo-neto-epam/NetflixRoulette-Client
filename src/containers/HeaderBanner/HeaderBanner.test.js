import React from 'react'
import '@testing-library/react'
import renderer from 'react-test-renderer';
import HeaderBanner from './index'

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

test('HeaderBanner renders correctly', () => {
    const tree = renderer
        .create(<HeaderBanner />)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });