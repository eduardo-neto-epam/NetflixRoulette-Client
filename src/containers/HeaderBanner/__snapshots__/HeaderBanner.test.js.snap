// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBanner renders correctly 1`] = `
Array [
  <div
    className="image-frame"
  >
    <div
      className="bg-image"
    />
  </div>,
  <div
    className="container"
  >
    <div
      className="container"
    >
      <div>
        <span
          className="brand brand--bold"
        >
          netflix
        </span>
        <span
          className="brand"
        >
          roulette
        </span>
      </div>
      <button
        className="btn btn--lg btn--glass"
        onClick={[Function]}
        type="button"
      >
        + ADD MOVIE
      </button>
    </div>
    <h1
      className="heading"
    >
      FIND YOUR MOVIE
    </h1>
    <div
      className="container"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="What do you want to watch?"
          required={true}
          value=""
        />
        <button
          className="btn btn--lg btn--primary"
          onClick={[Function]}
          type="submit"
        >
          SEARCH
        </button>
      </form>
    </div>
  </div>,
]
`;
